packages/near-fast-auth-signer/src/components/SignMultichain/utils.ts

```diff
@@ -167,31 +205,41 @@ export const shortenAddress = (address: string): string => {
 };
 
 export const multichainSignAndSend = async ({
-  domain,
-  asset,
-  to,
-  value,
+  signMultichainRequest,
   feeProperties
 }: {
-  domain: string;
-  asset: Chain;
-  to: string;
-  value: string;
+  signMultichainRequest: EvmSendMultichainMessage | BTCSendMultichainMessage;
   feeProperties: TransactionFeeProperties;
 }) => {
   const accountId = window.fastAuthController.getAccountId();
   const keypair = await window.fastAuthController.getKey(accountId);
-  const derivedPath = `,${asset},${domain}`;
+  const chainDetails = {
+    chain:   signMultichainRequest.chain,
+    chainId: (signMultichainRequest as EvmSendMultichainMessage).chainId,
+  };
+  const derivedPath = canonicalize(pickBy({
+    chain:  signMultichainRequest.chain,
+    domain: signMultichainRequest.domain,
+    meta:   signMultichainRequest.meta,
+  }, (v) => v !== undefined && v !== null));
+
   const chainConfig = {
     contract:    MULTICHAIN_CONTRACT_TESTNET,
-    ...CHAIN_CONFIG[asset],
+    ...CHAIN_CONFIG[getTokenSymbol(chainDetails)],
   };
-
-  if (isEVMChain(asset)) {
+  // chain is the slip044 chain id
+  if (chainDetails.chain === 60) {
```

Comment:
Same as the previous comment, properly type it to get errors in case we add/remove chain and forget to update here